kcidb/test_oo.py

```diff
@@ -264,6 +284,18 @@ def assertContains(self, container, contents):
 
                 self.assertContains(container[key], contents[key])
 
+    def get_valid_item(self, pattern_str, type):
```

Comment:
This mostly does `query_str` and then filters out the valid items. So we should either make sure the querying is obvious from the function name, and call it something like `query_str_valid()`, or just make a static method for filtering the result of invoking `query_str()`, call it something like `filter_valid()`, and change the invocation appropriately.